tools/myst/enc/enc.c

```diff
@@ -116,6 +116,9 @@ static uint64_t _vectored_handler(oe_exception_record_t* er)
                 er->context->rcx = rcx;
                 er->context->rdx = rdx;
 
+                /* Skip over the illegal instruction. */
+                er->context->rip += 2;
```

Comment:
I wonder why previously the `cpuid` test could pass without advancing the rip. Any explanation there?

Reply:
When RIP was not advanced by mistake, the CPUID instruction will be executed again with RAX value returned by the myst_cpuid_ocall. If the RAX value returned is one of the value in [0,1,4,7,0x80000000,0x80000001], the resulting CPUID instruction exception will be handled by the OE code, instead of the code in tools/myst/enc/enc.c. The OE code advances the RIP after emulating the CPUID result for [0,1,4,7,0x80000000,0x80000001].